File: Python3.x/228-Summary_Ranges.py
# Runtime: 44 ms, faster than 6.58% of Python3 online submissions for Summary Ranges.
# Memory Usage: 13.9 MB, less than 16.67% of Python3 online submissions for Summary Ranges.
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        n = len(nums)
        if n == 0:
            return []
        elif n == 1:
            return [str(nums[0])]
        l, r = 0, 1
        res = []
        while r <= n:
            while r < n and nums[r] == nums[r - 1] + 1:
                r += 1
            if r == l + 1:
                res.append(str(nums[l]))
            else:
                res.append(str(nums[l]) + "->" + str(nums[r - 1]))
            l = r
            r += 1
        return res

File: Algorithms/C++/228-Summary_Ranges.cpp
// Runtime: 4 ms, faster than 100.00% of C++ online submissions for Summary Ranges.
// Memory Usage: 6.9 MB, less than 55.49% of C++ online submissions for Summary Ranges.

class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        if (nums.size() == 0) {
            return {};
        }
        if (nums.size() == 1) {
            return { std::to_string(nums[0]) };
        }
        std::vector<string> res;
        int left = 0, right = 1;
        while (right <= nums.size()) {
            while (right < nums.size() && nums[right] == nums[right - 1] + 1) {
                right++;
            }
            if (left + 1 == right) {
                res.push_back(std::to_string(nums[left]));
            } else {
                res.push_back(std::to_string(nums[left]) + "->" 
                              + std::to_string(nums[right - 1]));
            }
            left = right;
            right++;
        }
        return res;
    }
};
